dc/core/ZoomInteractionDelegate.cpp

```diff
@@ -162,9 +162,16 @@ QSizeF ZoomInteractionDelegate::_getMaxZoom() const
 
 QSizeF ZoomInteractionDelegate::_getMinZoom() const
 {
-    // The aspect ratio of the zoom rectangle must match that of the window
     const QSizeF window( getWindowCoord().size( ));
-    return window.scaled( MIN_ZOOM, MIN_ZOOM, Qt::KeepAspectRatio );
+    QSizeF content( _contentWindow.getContent()->getDimensions( ));
+    content.scale( window, Qt::KeepAspectRatioByExpanding );
+    const qreal windowAR = window.width() / window.height();
+    const qreal contentAR = content.width() / content.height();
+
+    if( contentAR > windowAR )
+        return QSizeF( MIN_ZOOM * window.width() / content.width(), MIN_ZOOM );
+    else
```

Comment:
done, even if I still find it more clear to read an if / else in a case like this ;-)